src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/DeleteAllButton.java

```diff
@@ -0,0 +1,40 @@
+package com.wikia.webdriver.elements.mercury.components.discussions.common;
+
+import com.wikia.webdriver.pageobjectsfactory.pageobject.BasePageObject;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.TimeoutException;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+
+public class DeleteAllButton extends BasePageObject {
+
+  @FindBy(className = "delete-all")
+  private WebElement button;
+
+  // context of DeleteAllButton in which it should be displayed
+  @FindBy(css = ".discussion-page.row > div:nth-child(3).rail")
+  private WebElement rightRail;
+
+  public boolean isVisible() {
+    try {
+      return button.isDisplayed();
+    } catch (NoSuchElementException e) {
+      return false;
+    }
+  }
+
+  public boolean isNotVisible() {
+    try {
+      return rightRail.isEnabled() && wait.forElementNotVisible(button);
+    } catch (TimeoutException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)